src/nxt_status.c

```diff
@@ -6,18 +6,60 @@
 #include <nxt_main.h>
 #include <nxt_conf.h>
 #include <nxt_status.h>
+#include <nxt_application.h>
+
+
+nxt_inline const nxt_str_t *
+nxt_status_app_type2name(nxt_app_type_t type)
+{
+#define nxt_str_sconst(s)                                                     \
+    ({                                                                        \
+        static const nxt_str_t _scstr = nxt_string(s);                        \
+        &_scstr;                                                              \
+     })
+
+    switch (type) {
+    case NXT_APP_EXTERNAL:
+        return nxt_str_sconst("external");
+    case NXT_APP_PYTHON:
+        return nxt_str_sconst("python");
+    case NXT_APP_PHP:
+        return nxt_str_sconst("php");
+    case NXT_APP_PERL:
+        return nxt_str_sconst("perl");
+    case NXT_APP_RUBY:
+        return nxt_str_sconst("ruby");
+    case NXT_APP_JAVA:
+        return nxt_str_sconst("java");
+    case NXT_APP_WASM:
+        return nxt_str_sconst("wasm");
+    case NXT_APP_WASM_WC:
+        return nxt_str_sconst("wasm-wasi-component");
+    case NXT_APP_UNKNOWN:
+        return nxt_str_sconst("unknown");
+    }
+
+    /* Avoid 'warning: control reaches end of non-void function' */
+    nxt_unreachable();
+
+#undef nxt_str_sconst
+}
 
 
 nxt_conf_value_t *
-nxt_status_get(nxt_status_report_t *report, nxt_mp_t *mp)
+nxt_status_get(nxt_task_t *task, nxt_status_report_t *report, nxt_mp_t *mp)
```

Comment:
Good stuff.

We could get the job of getting the module name done earlier.
The modules/languages are discovered in the `discover` process which exits after finishing its job.
I feel the module name can be set in it.
1. set module name in `nxt_discovery_module()`.
2. then it passed to the main process, `nxt_main_port_modules_handler()`.
    I feel it makes sense to add a field to save the module name in `nxt_app_lang_module_t`.
3. use the module name directly in the status API.

Btw, you can get thread with `nxt_thread()` without adding `task` parameter.

Reply:
Sure, this is what I wrote in the commit message...

```
Due to the module name not actually being kept around (it could be added
but would be a much more intrusive patch) we need to map module types to
their name.
```

Do you want the more intrusive patch?

Reply:
Hmm, do JSON objects at the same level have to have unique names?...

Reply:
Oh, I didn't notice you had the idea.
I feel it's not more intrusive. Before the module name wasn't used, now it looks good to be added before control and router processes are created.
But it's just an idea, how good depends on the code.

Reply:
Oh, I didn't notice you had the idea.
I feel it's not more intrusive. Before the module name wasn't used, now it looks good to be added before control and router processes are created.
It just intuitively makes more sense, how good depends on the code.

Reply:
OK, so it's not _that_ bad... <https://git.digital-domain.net/unit.git/commit/?h=status-modules&id=d03b0d2cf431eceb3dd5ace97460749a1f1096f9>